snap-server.cabal

```diff
@@ -244,7 +244,8 @@ Test-suite testsuite
     parallel                            >= 3        && < 4,
     test-framework                      >= 0.8.0.3  && < 0.9,
     test-framework-hunit                >= 0.2.7    && < 0.4,
-    test-framework-quickcheck2          >= 0.2.12.1 && < 0.4
+    test-framework-quickcheck2          >= 0.2.12.1 && < 0.4,
+    Cabal
```

Comment:
Why is this new dependency here?  I don't see any code changes.

Reply:
IIRC https-streams complained about missing cabal lib. I've retested and the result of:
```
cabal sandbox init
cabal install --dependencies-only --enable-tests -fbuild-testserver -fopenssl
```
is this error message:
```
Configuring http-streams-0.8.5.5...
Failed to install http-streams-0.8.5.5
Build log ( /home/karel/git/snap-server/.cabal-sandbox/logs/http-streams-0.8.5.5.log ):
cabal: Error: some packages failed to install:
http-streams-0.8.5.5 failed during the configure step. The exception was:
user error (The package 'http-streams' requires Cabal library version -any &&
>=1.24 but no suitable version is installed.)
```
but well, indeed, adding Cabal there does not solve the issue. I've needed to perform separate
```
cabal install Cabal==2.0.1.1
```
to fix this and complete installation of dependencies. Then configure and cabal build run fine.
So I'll commit cleanup in a minute. Sorry for this and thanks for catching this.